### PR TITLE
Dev example mtls

### PR DESCRIPTION
**Pull request by otaviolcs3**

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# C-spiffe Examples
+
+This section contains a set of standalone examples that demonstrate different use cases for the go-spiffe library.
```
Comment (arianalima) [inline examples/README.md:3]:
```suggestion
This section contains a set of standalone examples that demonstrate different use cases for the c-spiffe library.
```

File: examples/spiffe-tls/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Mutually Authenticated TLS (mTLS)
+
+This example shows how to use the C-spiffe library to establish an mTLS connection between two workloads using X.509 SVIDs obtained from the SPIFFE Workload API.
```
Comment (arianalima) [inline examples/spiffe-tls/README.md:3]:
```suggestion
This example shows how to use the C-spiffe library to establish a mTLS connection between two workloads using X.509 SVIDs obtained from the SPIFFE Workload API. 
```

File: examples/spiffe-tls/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Mutually Authenticated TLS (mTLS)
+
+This example shows how to use the C-spiffe library to establish an mTLS connection between two workloads using X.509 SVIDs obtained from the SPIFFE Workload API. 
+
+One workload acts as a client and the other as the server. 
+
+The scenario goes like this:
+1. The server starts listening for incoming SPIFFE-compliant mTLS connections.
+2. The client establishes an SPIFFE-compliant mTLS connection to the server.
```
Comment (arianalima) [inline examples/spiffe-tls/README.md:9]:
```suggestion
2. The client establishes a SPIFFE-compliant mTLS connection to the server. 
```

**Review by arianalima — CHANGES_REQUESTED** [review]
Some typos

**Review by tjamir — APPROVED** [review]

File: examples/spiffe-tls/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Client: 
+```bash
+spire-server entry create -spiffeID spiffe://example.org/client \
+                          -parentID spiffe://example.org/host \
+                          -selector unix:user:root
+```
+
+### 2. Start the server
+Start the server with the `server-workload` user:
```
Comment (arianalima) [inline examples/spiffe-tls/README.md:73]:
```suggestion
```

File: examples/spiffe-tls/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+```bash
+./spiffe_tls_server
+```
+
+### 3. Run the client
+Run the client with the `client-workload` user:
```
Comment (arianalima) [inline examples/spiffe-tls/README.md:79]:
```suggestion
```

File: examples/spiffe-tls/README.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+```bash
+./spiffe_tls_client
+```
+
+The server should have received a _"Hello server"_ message and responded with a _"Hello client"_ message.
+
+If either workload encounters a peer with a different SPIFFE ID than the one it expects, the workload aborts the TLS handshake and the connection fails.
```
Comment (arianalima) [inline examples/spiffe-tls/README.md:86]: I'm not sure it still happens that way

**Review by arianalima — CHANGES_REQUESTED** [review]
some inconsistencies in the README file

**Review by arianalima — APPROVED** [review]

**Pull request merged**